src/Timecode.php

```diff
@@ -0,0 +1,375 @@
+<?php
+
+namespace FireworkWeb\SMPTE;
+
+/**
+ * Timecode class representation
+ */
+class Timecode
+{
+    /**
+     * @var float
+     */
+    const DEFAULT_FRAMERATE = 24.0;
+
+    /**
+     * @var bool
+     */
+    const DEFAULT_DROPFRAME = false;
+
+    /**
+     * @var int
+     */
+    const DROP_FRAME_CONCEPT = 17982;
+
+    /**
+     * @var float
+     */
+    private $frameRate;
+
+    /**
+     * @var bool
+     */
+    private $dropFrame;
+
+    /**
+     * @var int
+     */
+    private $frameCount;
+
+    /**
+     * @var int
+     */
+    private $hours;
+
+    /**
+     * @var int
+     */
+    private $minutes;
+
+    /**
+     * @var int
+     */
+    private $seconds;
+
+    /**
+     * @var int
+     */
+    private $frames;
+
+    /**
+     * Create a new Timecode
+     *
+     * @param int|string|DateTime $time
+     * @param float $frameRate
+     * @param bool $dropFrame
+     * @return void
+     */
+    public function __construct(
+        $time,
+        float $frameRate = self::DEFAULT_FRAMERATE,
+        bool $dropFrame = self::DEFAULT_DROPFRAME
+    ) {
+        if (! Validations::isValidTimeCode($time, $frameRate, $dropFrame)) {
+            throw new \InvalidArgumentException('Invalid timecode');
+        }
+
+        $this->frameRate = $frameRate;
+        $this->dropFrame = $dropFrame;
+
+        $this->generateFrameCount($time);
+    }
+
+    /**
+     * Convert the Timecode to its string representation.
+     *
+     * @return string
+     */
+    public function __toString() : string
+    {
+        return $this->toString();
+    }
+
+    /**
+     * Calculate the frame count based on time code
+     *
+     * @param string $time
+     * @param float $frameRate
+     * @param bool $dropFrame
+     * @return int
+     */
+    public static function frameCountFromTimecode(
+        $time,
+        float $frameRate = self::DEFAULT_FRAMERATE,
+        bool $dropFrame = self::DEFAULT_DROPFRAME
+    ) : int {
+        if (! Validations::isValidString($time, $frameRate, $dropFrame)) {
+            throw new \InvalidArgumentException('Invalid string format');
+        }
+
+        list($hours, $minutes, $seconds, $frames) = preg_split('/(:|;)/', $time);
+
+        $roundFrameRate = round($frameRate);
+        $frameCount = ($roundFrameRate * 60 * 60 * $hours)
```

Comment:
what would make things easier to understand and maintain is creating private/internal functions for links like this:

`$frameCount = $this->calculateFrameCount($frameRate, $hours, $minutes, $seconds, $frames)`.

Not required to change, just point for long term code, things that aren't really required to be on this method can be moved away